src/registry/domain/s3.js

```diff
@@ -11,13 +11,19 @@ const _ = require('lodash');
 const getFileInfo = require('../../utils/get-file-info');
 const getNextYear = require('../../utils/get-next-year');
 const strings = require('../../resources');
+const proxy = require('proxy-agent');
 
 module.exports = function(conf) {
+  const httpOptions = { timeout: conf.s3.timeout || 10000 };
+  if (conf.s3.agentProxy) {
+    httpOptions.agent = proxy(conf.s3.agentProxy);
```

Comment:
I'm a bit confused. If the property would be called `conf.s3.proxy` I would imagine it would be consistent to assign to the `httpOptions.agent` the `proxy(url)` value, but given we called it `agentProxy` I thought it was part of the idea to pass the whole agent already?

I think this would be a bit better given the naming, and also to allow who's not interested to use the proxy to avoid installing the `proxy-agent` dependency.

What's your thoughts on that @mattiaerre @NapoleanReigns

Reply:
I completely agree with it. We shouldn't mandate the proxy-agent dependency to the user unless he is in need of it.

I will make the changes and submit soon.